### PR TITLE
Set version label based on .Status.Version.Version from EdgeConnect

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

This PR use `.Status.Version.Version` as version label ~otherwise fallback to "latest".~

## How can this be tested?

Unit tests has been added.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by luhi-DT — APPROVED** [review]

**Review by luhi-DT — CHANGES_REQUESTED** [review]
As discussed needs rework

**Comment by 0sewa0** [comment]
I wouldn't call this a hotfix, and we don't really have hotfixes as that would require us to release it quick which we haven't done so far.
This is just a normal PR, as the thing we are fixing was never released and caused no actual issues

**Comment by andriisoldatenko** [comment]
> I wouldn't call this a hotfix, and we don't really have hotfixes as that would require us to release it quick which we haven't done so far. This is just a normal PR, as the thing we are fixing was never released and caused no actual issues

@0sewa0 fixed description.

File: src/controllers/edgeconnect/deployment/deployment.go [hunk 1]
```diff
@@ -88,11 +89,18 @@ func prepareContainerEnvVars(instance *edgeconnectv1alpha1.EdgeConnect) []corev1
 }
 
 func buildAppLabels(instance *edgeconnectv1alpha1.EdgeConnect) *kubeobjects.AppLabels {
+	version := "latest"
+	statusVersion := instance.Status.Version.Version
+
+	if statusVersion != "" && len(validation.IsQualifiedName(statusVersion)) == 0 {
```
Comment (chrismuellner) [inline src/controllers/edgeconnect/deployment/deployment.go:95]: We should just take whatever is in the status, and not validate it again. Similar with the empty check, we should default the version status to `latest` and use it as tag verbatim, without introducing duplicate logic here.

Comment (andriisoldatenko) [reply src/controllers/edgeconnect/deployment/deployment.go:95]: But it's not implemented, I created a ticket for that, we don't have code which reads LABEL.

**Review by 0sewa0 — APPROVED** [review]

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**